gpflow/conditionals.py

```diff
@@ -182,13 +214,15 @@ def _sample_conditional(Xnew, X, kern, f, *, q_sqrt=None, white=False, full_cov=
     samples = _sample_mvn(mean, cov, cov_structure, num_samples=num_samples)
     if full_cov:
         samples = tf.matrix_transpose(samples)
+        mean = tf.matrix_transpose(mean)
```

Comment:
this adds an unnecessary matrix_transpose ... would it be worth generating mean_PN instead? (could be mean_PN = mean in one branch and mean_PN = tf.matrix_transpose(mean) in the other)

Reply:
Yes I agree

Reply:
Though a transpose is quite cheap

Reply:
I wouldn't be so confident! Transpose is cheap in numpy, not in TensorFlow. Requires confirmation though.

Reply:
I've seen cases where transposes were actually one of the bottlenecks - can be a lot of memory to copy around ...

Reply:
As far as I remember transpose op in tensorflow is simple permutation of elements, therefore memory allocations are involved.